src/core/CCArray.mli

```diff
@@ -130,97 +147,103 @@ val sort_ranking : ('a -> 'a -> int) -> 'a t -> int array
 val find_map : ('a -> 'b option) -> 'a t -> 'b option
 (** [find_map f a] returns [Some y] if there is an element [x] such
     that [f x = Some y]. Otherwise returns [None].
-    @since 1.3 *)
+    @since 1.3, but only
+    @since 2.1 with labels *)
 
 val find : ('a -> 'b option) -> 'a t -> 'b option
 (** [find f a] is an alias to {!find_map}.
-    @deprecated since 1.3, use {!find_map} instead. *)
+    @deprecated since 1.3, use {!find_map} instead.
+    The version with labels is
+    @deprecated since 2.1, use {!find_map} instead. *)
 
 val find_map_i : (int -> 'a -> 'b option) -> 'a t -> 'b option
 (** [find_map_i f a] is like {!find_map}, but the index of the element is also passed
     to the predicate function [f].
-    @since 1.3 *)
+    @since 1.3, but only
+    @since 2.1 with labels *)
 
 val findi : (int -> 'a -> 'b option) -> 'a t -> 'b option
 (** [findi f a] is an alias to {!find_map_i}.
     @since 0.3.4
-    @deprecated since 1.3, use {!find_map_i} instead. *)
+    @deprecated since 1.3, use {!find_map} instead.
+    The version with labels is
+    @deprecated since 2.1, use {!find_map} instead. *)
 
 val find_idx : ('a -> bool) -> 'a t -> (int * 'a) option
-(** [find_idx p a] returns [Some (i,x)] where [x] is the [i]-th element of [a],
-    and [p x] holds. Otherwise returns [None].
+(** [find_idx f a] returns [Some (i,x)] where [x] is the [i]-th element of [a],
+    and [f x] holds. Otherwise returns [None].
     @since 0.3.4 *)
 
-val lookup : cmp:'a ord -> 'a -> 'a t -> int option
-(** [lookup cmp x a] lookups the index of some key [x] in a sorted array [a].
-    Undefined behavior if the array [a] is not sorted wrt [cmp].
+val lookup : cmp:('a ord [@keep_label]) -> 'a -> 'a t -> int option
+(** [lookup ~cmp key a] lookups the index of some key [key] in a sorted array [a].
+    Undefined behavior if the array [a] is not sorted wrt [~cmp].
     Complexity: [O(log (n))] (dichotomic search).
-    @return [None] if the key [x] is not present, or
+    @return [None] if the key [key] is not present, or
       [Some i] ([i] the index of the key) otherwise. *)
 
-val lookup_exn : cmp:'a ord -> 'a -> 'a t -> int
-(** [lookup_exn cmp x a] is like {!lookup}, but
-    @raise Not_found if the key [x] is not present. *)
+val lookup_exn : cmp:('a ord [@keep_label]) -> 'a -> 'a t -> int
+(** [lookup_exn ~cmp key a] is like {!lookup}, but
+    @raise Not_found if the key [key] is not present. *)
 
-val bsearch : cmp:('a -> 'a -> int) -> 'a -> 'a t ->
+val bsearch : cmp:(('a -> 'a -> int) [@keep_label]) -> 'a -> 'a t ->
```

Comment:
I'm afraid the `keep_label` attribute might break on 4.02 :-/
It seems the inline attribute is already problematic…

We may need some kind of AST preprocessor which depends on OCaml's version?

Reply:
I'll test on 4.02. This label is obviously by easily removed automatically.

Reply:
Tested and doesn't break 4.02.3.
It seems like 4.02 not only supports inline attributes for expressions, but also for type expressions :smile: